### PR TITLE
Make Transit key creation return an Optional key to support previous Vault versions

### PR DESCRIPTION
**Pull request by kdubb**

### CODE REVIEW

**Comment by kdubb** [comment]
@vsevel This should fix #249

**Comment by kdubb** [comment]
I made it return an Optional as discussed.

File: client/src/main/specs/secrets/transit.yaml [hunk 1]
```diff
@@ -32,7 +33,8 @@ operations:
       type: String
   result:
     kind: leased
-    unwrapData: true
+    unwrapUsing: "r == null ? Optional.empty() : Optional.of(r.getData())"
```
Comment (vsevel) [inline client/src/main/specs/secrets/transit.yaml:36]: `Optional.ofNullable(r)`

**Review by vsevel — APPROVED** [review]

File: client/src/main/specs/secrets/transit.yaml (hunk 1 above)
Comment (kdubb) [reply client/src/main/specs/secrets/transit.yaml:36]: Good catch

**Pull request merged**